trunner/target/armv7m4.py

```diff
@@ -24,9 +25,26 @@
 
 
 class ARMv7M4Rebooter(Rebooter):
-    # TODO add text mode
-    # NOTE: changing boot modes not needed/supported for this target
-    pass
+    def _reboot_dut_text(self, hard):
```

Comment:
According to our last meeting it will be better to add another method, for example called `_reboot_dut_command`, because as you can see sending a reset command to onboard debugger is definitely not a text mode.
```suggestion
    def _reboot_dut_command(self, hard):
```

Then we should introduce a launch_context, where we could specify what reset do we want to use. For now we can just change it temporary this way if needed:
![Screenshot from 2024-04-15 12-24-41](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/77304431/63e398eb-0482-492f-8d18-6536f6043555)